### PR TITLE
Add API docs for the FluxInstance custom resource

### PR DESCRIPTION
**Pull request by stefanprodan**
Add docs for the FluxInstance API spec and status.

### CODE REVIEW

File: docs/api/v1/fluxinstance.md [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+- `type: Ready`
+- `status: "False"`
+- `reason: BuildFailed | HealthCheckFailed | ReconciliationFailed`
+
+The `message` field of the Condition will contain more information about why
+the reconciliation failed.
+
+While the FluxInstance has one or more of these Conditions, the flux-operator
+will continue to attempt a reconciliation with an
+exponential backoff, until it succeeds and the FluxInstance marked as [ready](#ready-kustomization).
```
Comment (souleb) [inline docs/api/v1/fluxinstance.md:397]:
```suggestion
exponential backoff, until it succeeds and the FluxInstance is marked as [ready](#ready-kustomization).
```

File: docs/api/v1/fluxinstance.md [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+
+   ```console
+    $ kubectl -n flux-system get fluxinstance
+    NAME   AGE   READY   STATUS                           REVISION
+    flux   59s   True    Reconciliation finished in 52s   v2.3.0@sha256:4cc5babdb1279ad0177bf513292deadbfa3f7b7c3da0be7fa53b39ab434f7219
+   ```
+
+3. Run `kubectl describe fluxinstance` to see the reconciliation status components, conditions and events:
+
+    ```console
```
Comment (souleb) [inline docs/api/v1/fluxinstance.md:78]:
```suggestion
```console
```

File: docs/api/v1/fluxinstance.md [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+        Status:                True
+        Type:                  Ready
+      Last Applied Revision:    v2.3.0@sha256:4cc5babdb1279ad0177bf513292deadbfa3f7b7c3da0be7fa53b39ab434f7219
+      Last Attempted Revision:  v2.3.0@sha256:4cc5babdb1279ad0177bf513292deadbfa3f7b7c3da0be7fa53b39ab434f7219
+    Events:
+      Type    Reason                   Age    From             Message
+      ----    ------                   ----   ----             -------
+      Normal  Progressing              6m20s  flux-controller  Installing revision v2.3.0@sha256:4cc5babdb1279ad0177bf513292deadbfa3f7b7c3da0be7fa53b39ab434f7219
+      Normal  ReconciliationSucceeded  5m9s   flux-controller  Reconciliation finished in 52s
+    ```
```
Comment (souleb) [inline docs/api/v1/fluxinstance.md:120]:
```suggestion
```
```

File: docs/api/v1/fluxinstance.md [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+
+4. Run `kubectl logs` on the flux-operator pod to see the reconciliation logs:
+
+    ```shell
```
Comment (souleb) [inline docs/api/v1/fluxinstance.md:124]:
```suggestion
```shell
```

File: docs/api/v1/fluxinstance.md [hunk 5]
```diff
@@ -0,0 +1,466 @@
...
+    kubectl -n flux-system logs deployment/flux-operator
+    ```
```
Comment (souleb) [inline docs/api/v1/fluxinstance.md:126]:
```suggestion
```
```

File: docs/api/v1/fluxinstance.md [hunk 6]
```diff
@@ -0,0 +1,466 @@
...
+```
+
+### Inventory status
+
+In order to perform operations such as drift detection, garbage collection, upgrades, etc.,
+the flux-operator needs to keep track of all Kubernetes objects that are
+reconciled as part of a FluxInstance. To do this, it maintains an inventory
+containing the list of Kubernetes resource object references that have been
+successfully applied and records it in `.status.inventory`. The inventory
+records are in the format `<namespace>_<name>_<group>_<kind>_<version>`.
```
Comment (souleb) [inline docs/api/v1/fluxinstance.md:427]:
```suggestion
records are in the format `Id: <namespace>_<name>_<group>_<kind>, V:<version>`.
```

File: docs/api/v1/fluxinstance.md (hunk 2 above)
Comment (stefanprodan) [reply docs/api/v1/fluxinstance.md:78]: The indentation is needed for the numbered list

**Review by souleb — APPROVED** [review]
LGTM!

**Pull request merged**